sdk/typespec/typespec_client_core/src/http/context.rs

```diff
@@ -15,6 +15,16 @@ pub struct Context<'a> {
     type_map: Cow<'a, HashMap<TypeId, Arc<dyn Any + Send + Sync>>>,
 }
 
+impl Context<'static> {
+    /// Creates a new [`Context`] that borrows from the current [`Context`].
+    pub fn to_borrowed(&self) -> Context<'_> {
+        let type_map = self.type_map.as_ref();
+        Context {
+            type_map: Cow::Borrowed(type_map),
+        }
+    }
```

Comment:
Given this is what `with_context` does, you might just call that or at least use the same code. Or maybe that calls this. Just want to avoid duplication as much as possible.

I'm fine adding this, though, even though they do the same thing because this is more obvious. The other was meant more for construction. Or maybe we deprecate `with_context` in favor of this one.

Reply:
That's a good point, I had actually completely missed `with_context` 🤦🏻‍♀️ . Given that we use `with_context` heavily already and it works fine for my case, I'm actually ok just removing this for now. If we do want to switch to something called as a method instead of a constructor, I'm quite happy to do that as a separate refactoring.